src/core/function.jl

```diff
@@ -39,7 +39,7 @@ struct FunctionAttrSet
     idx::API.LLVMAttributeIndex
 end
 
-function_attributes(f::Function) = FunctionAttrSet(f, API.LLVMAttributeFunctionIndex)
+function_attributes(f::Function) = FunctionAttrSet(f, reinterpret(API.LLVMAttributeIndex, API.LLVMAttributeFunctionIndex))
```

Comment:
This change should be reviewed before merging.

Reply:
That looks OK.